samples/bluetooth/central_hr/sysbuild.cmake

```
@@ -2,23 +2,55 @@
 # SPDX-License-Identifier: Apache-2.0
 
 if(SB_CONFIG_NET_CORE_IMAGE_HCI_IPC)
-	# For builds in the nrf5340, we build the netcore image with the controller
+  # For builds in the nrf5340, we build the netcore image with the controller
+  set(NET_APP hci_ipc)
+  set(NET_APP_SRC_DIR ${ZEPHYR_BASE}/samples/bluetooth/${NET_APP})
 
-	set(NET_APP hci_ipc)
-	set(NET_APP_SRC_DIR ${ZEPHYR_BASE}/samples/bluetooth/${NET_APP})
+  ExternalZephyrProject_Add(
+    APPLICATION ${NET_APP}
+    SOURCE_DIR  ${NET_APP_SRC_DIR}
+    BOARD       ${SB_CONFIG_NET_CORE_BOARD}
+  )
 
-	ExternalZephyrProject_Add(
-		APPLICATION ${NET_APP}
-		SOURCE_DIR  ${NET_APP_SRC_DIR}
-		BOARD       ${SB_CONFIG_NET_CORE_BOARD}
-	)
+  set(${NET_APP}_CONF_FILE
+      ${NET_APP_SRC_DIR}/nrf5340_cpunet_iso-bt_ll_sw_split.conf
+      CACHE INTERNAL ""
+  )
 
-	set(${NET_APP}_CONF_FILE
-	 ${NET_APP_SRC_DIR}/nrf5340_cpunet_iso-bt_ll_sw_split.conf
-	 CACHE INTERNAL ""
-	)
+  if(SB_CONFIG_TEST_MODE_PHY_CODED)
+    set_config_bool(${NET_APP} CONFIG_BT_CTLR_PHY_CODED y)
+    set_config_bool(${NET_APP} CONFIG_BT_EXT_ADV y)
+  elseif(SB_CONFIG_TEST_MODE_EXTENDED)
+    set_config_bool(${NET_APP} CONFIG_BT_CTLR_PHY_CODED n)
+    set_config_bool(${NET_APP} CONFIG_BT_EXT_ADV y)
+  else()
+    set_config_bool(${NET_APP} CONFIG_BT_CTLR_PHY_CODED n)
+    set_config_bool(${NET_APP} CONFIG_BT_EXT_ADV n)
+  endif()
 
-	native_simulator_set_child_images(${DEFAULT_IMAGE} ${NET_APP})
+  if(SB_CONFIG_TEST_MODE_EXTENDED OR SB_CONFIG_TEST_MODE_PHY_CODED)
+    set_config_int(${NET_APP} CONFIG_BT_CTLR_SCAN_DATA_LEN_MAX ${SB_CONFIG_BT_CTLR_SCAN_DATA_LEN_MAX})
+    set_config_int(${NET_APP} CONFIG_BT_BUF_EVT_DISCARDABLE_SIZE ${SB_CONFIG_BT_BUF_EVT_DISCARDABLE_SIZE})
+  endif()
+
+  native_simulator_set_child_images(${DEFAULT_IMAGE} ${NET_APP})
 endif()
 
 native_simulator_set_final_executable(${DEFAULT_IMAGE})
+
+if(SB_CONFIG_BT_CTLR_PHY_CODED)
+  set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_CTLR_PHY_CODED y)
+  set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_EXT_ADV y)
+  set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_AUTO_PHY_UPDATE n)
+elseif(SB_CONFIG_TEST_MODE_EXTENDED)
+  set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_CTLR_PHY_CODED n)
+  set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_EXT_ADV y)
+else()
+  set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_CTLR_PHY_CODED n)
+  set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_EXT_ADV n)
+endif()
+
+if(SB_CONFIG_TEST_MODE_EXTENDED OR SB_CONFIG_TEST_MODE_PHY_CODED)
+  set_config_int(${DEFAULT_IMAGE} CONFIG_BT_CTLR_SCAN_DATA_LEN_MAX ${SB_CONFIG_BT_CTLR_SCAN_DATA_LEN_MAX})
+  set_config_int(${DEFAULT_IMAGE} CONFIG_BT_BUF_EVT_DISCARDABLE_SIZE ${SB_CONFIG_BT_BUF_EVT_DISCARDABLE_SIZE})
+endif()
```

Control KConfig not be supplied to the sample, samples shall not fail to build and execute of the above Controller build has the correct features.

```suggestion
if(SB_CONFIG_BT_CTLR_PHY_CODED)
  set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_EXT_ADV y)
  set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_AUTO_PHY_UPDATE n)
elseif(SB_CONFIG_TEST_MODE_EXTENDED)
  set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_EXT_ADV y)
else()
  set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_EXT_ADV n)
endif()

if(SB_CONFIG_TEST_MODE_EXTENDED OR SB_CONFIG_TEST_MODE_PHY_CODED)
  set_config_int(${DEFAULT_IMAGE} CONFIG_BT_BUF_EVT_DISCARDABLE_SIZE ${SB_CONFIG_BT_BUF_EVT_DISCARDABLE_SIZE})
endif()
```

---

Since the overlay files exists, i thought sysbuild is not required for nRF52/54 single core SoCs. Are we enforcing sysbuild for single core SoCs too?

---

We are not enforcing using sysbuild, but at this point we are supporting and using it in the tests. (Meaning, we just build with sysbuild, without distinction for either 52, 53 or 54L)

---

i am actually confused for the need of this PR to hav extensive changes, the central_hr and peripheral_hr is already being tested in CI for nrf52, nrf53 and nrf54 bsim targets using the overlays.

```
# west build --sysbuild -b nrf5340dk/nrf5340/cpuapp  -d build/central_hr samples/bluetooth/central_hr
```
```
# west build --sysbuild -b nrf5340dk/nrf5340/cpuapp  -d build/central_hr samples/bluetooth/central_hr -- -DEXTRA_CONF_FILE=overlay-extended.conf
```
```
# west build --sysbuild -b nrf5340dk/nrf5340/cpuapp  -d build/central_hr samples/bluetooth/central_hr -- -DEXTRA_CONF_FILE=overlay-phy_coded.conf
```

All the above build fine upstream. Except, that all the above have extended and coded PHY enabled in the hci_ipc; which means the extended and coded PHY variants both test coded PHY advertising and connection.

Just getting the hci_ipc build for `overlay-extended.conf` use in CPUAPP needs to set `CONFIG_BT_CTLR_PHY_CODED=n` for the hci_ipc. (CPUAPP samples are capable of detecting failure to use coded PHY and fallback to 2M PHY use).

---

I did updates to https://github.com/zephyrproject-rtos/zephyr/pull/75773 to correctly use sysbuild based on the changes present in that PR. So I'm not really sure what needs to be done here, can you explain what needs changing and what should be set (and what needs to be set in which image, for nrf52 it all goes to one image but in the case of e.g. nrf53, what config needs to go to which core?)